### PR TITLE
Prow secrets management docu

### LINKED ISSUE #28: Implement encrypting the secrets using KMS and storing them on GCS
Google does not provide Vault as a service. They provide KMS only for encrypting secrets but not storing. Encrypting using KMS and storing on GCS is the option.

Consider the secret format.
The goal is to read it out of GCS and create K8s Secret object(s) out of this data in a Prow cluser as simply as possible.

Two possible options for data format are: Storing everything in one encrypted yaml file (easier read/apply) vs storing every object separately (easier update).

AC:
- [ ] provide two credentials stored in GCS: service account to provision cluster for Kyma-Integration job, github account for github integration)
- [ ] document how to create initial credentials and update existing (a written procedure)

### PR DESCRIPTION
**Pull request by strekm**
**Description**

Changes proposed in this pull request:

- Prow secrets management docu

**Related issue(s)**
close #28

### CODE REVIEW

File: prow-configurations/README.md [hunk 1]
```diff
@@ -14,20 +14,68 @@ Install the following tools:
...
+```
+
+7. Encrypt the secret:
+```
+gcloud kms encrypt --location global --keyring prow-keyring --key prow-encryption-key --plaintext-file sa-gke-kyma-integration.json --ciphertext-file sa-gke-kyma-integration.json.encrypted
+```
+
+8. Upload encrypted secret to GCP:
+```
+gsutil cp sa-gke-kyma-integration.json.encrypted gs://mst-prow-bucket/
```
Comment (crabtree) [inline prow-configurations/README.md]:
```suggestion
gsutil cp sa-gke-kyma-integration.json.encrypted gs://{BUCKET_NAME}/
```

File: prow-configurations/prow-secrets-management.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+->**NOTE:** Before following this guide check Prow secrets setup for the $PROJECT_NAME.
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
+
+Run following command:
```
Comment (Tomasz-Smelcerz-SAP) [inline prow-configurations/prow-secrets-management.md]:
```suggestion
Run the following command:
```

File: prow-configurations/prow-secrets-management.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
+# Prow secrets management
+
+## Overview
+
+This document explains secrets management for Prow. Some jobs require using sensitive data. You need to encrypt data using KMS and store them in GCS. The following instructions assume that you are logged-in to the Google Cloud project with administrative rights.
+
+## Prerequisities
+
+ - [gcloud](https://cloud.google.com/sdk/gcloud/) installed.
```
Comment (jakkab) [inline prow-configurations/prow-secrets-management.md]:
```suggestion
 - [gcloud](https://cloud.google.com/sdk/gcloud/) 
```

File: prow-configurations/prow-secrets-management.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+For your convenience, export the following data:
+ - PROJECT_NAME - Google Cloud Project
+ - BUCKET_NAME - GCS bucket in the $PROJECT_NAME where Prow secrets are stored
+ - KEYRING_NAME - KMS key ring
+ - ENCRYPTION_KEY_NAME - key in the $KEYRING_NAME, used for data encryption
+
+## Secrets management
+
+->**NOTE:** Before following this guide check Prow secrets setup for the $PROJECT_NAME.
```
Comment (Tomasz-Smelcerz-SAP) [inline prow-configurations/prow-secrets-management.md]:
```suggestion
->**NOTE:** Perform each step listed below only if not done already.
```

File: prow-configurations/prow-secrets-management.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
+
```
Comment (Tomasz-Smelcerz-SAP) [inline prow-configurations/prow-secrets-management.md]:
```suggestion

The purpose of the bucket is to store encrypted credentials necessary for prow jobs like provisioning clusters/vms etc.
```

File: prow-configurations/prow-secrets-management.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
+# Prow secrets management
+
+## Overview
+
+This document explains secrets management for Prow. Some jobs require using sensitive data. You need to encrypt data using KMS and store them in GCS. The following instructions assume that you are logged-in to the Google Cloud project with administrative rights.
```
Comment (Tomasz-Smelcerz-SAP) [inline prow-configurations/prow-secrets-management.md]:
```suggestion
Some jobs require using sensitive data. You need to encrypt data using KMS and store them in GCS.
This document shows the commands necessary to create a service account and store it's encrypted key in GCS bucket. The following instructions assume that you are logged-in to the Google Cloud project with administrative rights.
```

File: prow-configurations/prow-secrets-management.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+Add policy binding for the $SA_NAME service account:
+```
+gcloud iam service-accounts add-iam-policy-binding $SA_NAME --member=serviceAccount:$SA_NAME@$PROJECT_NAME.iam.gserviceaccount.com --role=$ROLE
+```
+
+### Encrypt the secret
+
+Export following:
+ - KEYRING_NAME - KMS key ring name
```
Comment (Tomasz-Smelcerz-SAP) [inline prow-configurations/prow-secrets-management.md]: unnecessary

File: prow-configurations/prow-secrets-management.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+ - [gcloud](https://cloud.google.com/sdk/gcloud/) 
+ - basic knowledge about [GCP key rings and keys](https://cloud.google.com/kms/docs/creating-keys).
+
+For your convenience, export these data:
+ - PROJECT_NAME - Google Cloud Project
+ - BUCKET_NAME - GCS bucket in the $PROJECT_NAME where Prow secrets are stored
+ - KEYRING_NAME - KMS key ring
+ - ENCRYPTION_KEY_NAME - key in the $KEYRING_NAME, used for data encryption
+
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]:
```
Use the `export {VARIABLE}={value}` command to set up these variables, where:
 - **PROJECT_NAME** is a Google Cloud project.
 - **BUCKET_NAME** is a GCS bucket in the Google Cloud project that is used to store Prow secrets.
 - **KEYRING_NAME** is the KMS key ring.
 - **ENCRYPTION_KEY_NAME** is the key name in the key ring that is used for data encryption.
```

File: prow-configurations/prow-secrets-management.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+## Secrets management
+
+>**NOTE:** Before following this guide, check Prow secrets setup for the $PROJECT_NAME.
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Create GCS bucket -> Create a GCS bucket

File: prow-configurations/prow-secrets-management.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+ - PROJECT_NAME - Google Cloud Project
+ - BUCKET_NAME - GCS bucket in the $PROJECT_NAME where Prow secrets are stored
+ - KEYRING_NAME - KMS key ring
+ - ENCRYPTION_KEY_NAME - key in the $KEYRING_NAME, used for data encryption
+
+## Secrets management
+
+>**NOTE:** Before following this guide, check Prow secrets setup for the $PROJECT_NAME.
+
+Set context to the $PROJECT_NAME executing:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Execute this command to set the context to the Google Cloud project:

File: prow-configurations/prow-secrets-management.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+For your convenience, export these data:
+ - PROJECT_NAME - Google Cloud Project
+ - BUCKET_NAME - GCS bucket in the $PROJECT_NAME where Prow secrets are stored
+ - KEYRING_NAME - KMS key ring
+ - ENCRYPTION_KEY_NAME - key in the $KEYRING_NAME, used for data encryption
+
+## Secrets management
+
+>**NOTE:** Before following this guide, check Prow secrets setup for the $PROJECT_NAME.
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: - Before following -> Before you follow
- for the $PROJECT_NAME. -> for the Google Cloud project.

File: prow-configurations/prow-secrets-management.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
+
+The purpose of the bucket is to store encrypted credentials necessary for prow jobs like provisioning clusters/vms etc.
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: - prow -> Prow
- like provisioning clusters/vms etc. -> like provisioning clusters or virtual machines.

File: prow-configurations/prow-secrets-management.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+ - SECRET_FILE - path to the file containing secret
+
+Encrypt the secret:
+```
+gcloud kms encrypt --location global --keyring $KEYRING_NAME --key $ENCRYPTION_KEY_NAME --plaintext-file $SECRET_FILE --ciphertext-file $SECRET_FILE.encrypted
+```
+
+### Upload the secret
+
+Upload encrypted secret to GCP:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: encrypted secret -> the encrypted secret

File: prow-configurations/prow-secrets-management.md [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+Set context to the $PROJECT_NAME executing:
+```
+gcloud config set project $PROJECT_NAME
+```
+
+### Create GCS bucket
+
+The purpose of the bucket is to store encrypted credentials necessary for prow jobs like provisioning clusters/vms etc.
+Run this command:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Run this command to create a bucket:

File: prow-configurations/prow-secrets-management.md [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+```
+gsutil mb -p $PROJECT_NAME gs://$BUCKET_NAME/
+```
+
+### Create Google Service Account
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Create Google Service Account - Create a Google service account

File: prow-configurations/prow-secrets-management.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+Export these variables:
+ - SA_NAME - Service Account name
+ - SA_DISPLAY_NAME - Service Account display name
+ - SECRET_FILE - the path where the private key will be written
+ - ROLE - the role bound to the $SA_NAME
+
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]:
```
Export the variables, where:
 - **SA_NAME** is the name of the service account.
 - **SA_DISPLAY_NAME** is the display name of the service account.
 - **SECRET_FILE** is the path for the private key.
 - **ROLE** is the role bound to the service account.

```

File: prow-configurations/prow-secrets-management.md [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+Create a service account:
+```
+gcloud iam service-accounts create $SA_NAME --display-name $SA_DISPLAY_NAME
+```
+
+Create a private key for the $SA_NAME service account:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: for the $SA_NAME service account: -> for the service account:

File: prow-configurations/prow-secrets-management.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+```
+gcloud iam service-accounts keys create $SECRET_FILE --iam-account=SA_NAME
+```
+
+Add policy binding for the $SA_NAME service account:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Add a policy binding for the service account:

File: prow-configurations/prow-secrets-management.md [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+```
+gcloud iam service-accounts add-iam-policy-binding $SA_NAME --member=serviceAccount:$SA_NAME@$PROJECT_NAME.iam.gserviceaccount.com --role=$ROLE
+```
+
+### Encrypt the secret
+
+Export the following:
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Export the **SECRET_FILE** variable which is the path to the file containing the secret.

File: prow-configurations/prow-secrets-management.md [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+### Upload the secret
+
+Upload encrypted secret to GCP:
+```
+gsutil cp $SECRET_FILE.encrypted gs://$BUCKET_NAME/
+```
+
+Delete the SECRET_FILE file.
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: Delete the file exported under the **SECRET_FILE** variable.

File: prow-configurations/README.md [hunk 21]
```diff
@@ -4,6 +4,8 @@
 
 This folder contains the installation script and the set of configurations for Prow. 
 
+>**NOTE:** The following instructions assume that you are logged-in to the gcloud {PROJECT_NAME} with administrative rights.
```
Comment (kazydek) [inline prow-configurations/README.md]: logged-in to the gcloud {PROJECT_NAME} -> logged in to the gcloud project.

File: prow-configurations/prow-secrets-management.md [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+
+Export the following:
+ - SECRET_FILE - path to the file containing secret
+
+Encrypt the secret:
+```
+gcloud kms encrypt --location global --keyring $KEYRING_NAME --key $ENCRYPTION_KEY_NAME --plaintext-file $SECRET_FILE --ciphertext-file $SECRET_FILE.encrypted
+```
+
+### Upload the secret
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: secret -> Secret

File: prow-configurations/prow-secrets-management.md [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+```
+gcloud iam service-accounts keys create $SECRET_FILE --iam-account=SA_NAME
+```
+
+Add policy binding for the $SA_NAME service account:
+```
+gcloud iam service-accounts add-iam-policy-binding $SA_NAME --member=serviceAccount:$SA_NAME@$PROJECT_NAME.iam.gserviceaccount.com --role=$ROLE
+```
+
+### Encrypt the secret
```
Comment (kazydek) [inline prow-configurations/prow-secrets-management.md]: secret -> Secret

File: docs/prow-secrets-management.md [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+
+Export the following:
+ - SECRET_FILE - path to the file containing secret
+
+Encrypt the Secret:
+```
+gcloud kms encrypt --location global --keyring $KEYRING_NAME --key $ENCRYPTION_KEY_NAME --plaintext-file $SECRET_FILE --ciphertext-file $SECRET_FILE.encrypted
+```
+
+### Upload the secret
```
Comment (kazydek) [inline docs/prow-secrets-management.md]:
```suggestion
### Upload the Secret
```

**Pull request merged**